core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -371,11 +372,13 @@ class AdminManager(val config: KafkaConfig,
       try {
         val configEntriesMap = config.entries.asScala.map(entry => (entry.name, entry.value)).toMap
 
-        val configProps = new Properties
+        val configPropsOrigin = new Properties
         config.entries.asScala.foreach { configEntry =>
-          configProps.setProperty(configEntry.name, configEntry.value)
+          configPropsOrigin.setProperty(configEntry.name, configEntry.value)
         }
 
+        val configProps = resolveVariableConfigs(configPropsOrigin)
```

Comment:
Do we also want to include this resolving logic in the `IncrementalAlterConfigs` API?

Reply:
I reverted changes because this solution will write resolved config to ZK. I moved resolveVariableConfigs to DynamicBrokerConfig. So this will cover IncrementalAlterConfigs scenario that will call alterBrokerConfigs.